### PR TITLE
feat(pypi): auto PyPI release

### PR DESCRIPTION
**Pull request by fantix**
### New Features
* Auto PyPI release
* Auto release notes

### Dependency updates
* Use Pipenv for dev requirement management
* Switched to compatible versioning

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
The style in this PR agrees with [`black`](https://github.com/ambv/black). :heavy_check_mark:

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

File: README.md [hunk 1]
```diff
@@ -182,12 +182,14 @@ cirrus_config.update(**settings)
...
 
 ## Building the Documentation
-- `pip install -r dev-requirements.txt`
-- `python docs/create_docs.py`
+- `pipenv install --dev`
+- `pipenv run python docs/create_docs.py`
 - HTML is generated in the `docs/build` folder
 
-## Python 3 Compatibility Notes
-`psutil` doesn't install correctly when doing `pip install -r requirements.txt`
```
Comment (Avantol13) [inline README.md]: does this work with pipenv?

File: release_notes.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+pip install git+https://github.com/uc-cdis/release-helper.git@master#egg=gen3git
+gen3git --repo $TRAVIS_REPO_SLUG --from-tag $(git describe --tags --abbrev=0 --match=[0-9]* --exclude=$TRAVIS_TAG) gen --text --markdown --to-tag $TRAVIS_TAG
+URL=$(curl -s -H "Authorization:token $GH_TOKEN" https://api.github.com/repos/${TRAVIS_REPO_SLUG}/releases/tags/$TRAVIS_TAG | python -c "import sys, json; print(json.load(sys.stdin)['url'])")
```
Comment (Avantol13) [inline release_notes.sh]: how does `GH_TOKEN` get set? This is in travis env vars?

File: .travis.yml [hunk 3]
```diff
@@ -9,16 +11,37 @@ matrix:
    - python: "3.6"
 
 env:
-  - REPOSITORY="uc-cdis/cirrus" PR_NUMBER="$TRAVIS_PULL_REQUEST"
+  matrix:
+    - REPOSITORY="uc-cdis/cirrus" PR_NUMBER="$TRAVIS_PULL_REQUEST"
+  global:
+    secure: h7UxO6c+y15KjY+qbX2Ws0a9sF8vnuK1Y3arxgQV8VcVc0qF5iCHkGVfwTFgA00S69qvsGfU34kVjcpMABPqLR//J3e5WyJAhMP/NVJohDDzMvAw8DkbxsZmxuGIjqNG0Mo6cpwUG99bdDuo9yjZrVzK7cFSKr6F2EQWmAXMxezlLxck4VdavD/F5CHI/KXxzMmS6RIHUq/bTd37KVgqkPAIqkMnxWM0ciqwnz1SXK8XJSnDX/jVv2sEv5yWgAx4x/o3qBTKErY+J3yMr8ZD2FsXgsC79SwbjfwFIWWO02L1Fk6iHa3uWwrS+nr+O9aqZHaQz7Dr1TiPQnBe3y2mWtJ4Tzw5qDpBfIbobyLfAqx8Ur+h6OKf32Q2byloHc0jk9qXjSlEk9Z0D/hmUtjjejOhWClCNuh/5GHQ+oz2qjrOp+lo1CMfKwvaL5AvHkvtRIQzsMVTUzVSAYeSSB23vP3Z60B9ytjmcDeEuG9dkVoWd2O+57vHDXf94tF2TCBwITOlaa9mJ4VwvMjxA7EOjPAQpgyw+begrQzbnrE0ieF/nnI3dHEC3pcNFKKaE5ce+iisx4hA2eJpYaEiydc34wKnuT3z+J72lhop55SZNc49jYIAKTL8Weqel96t9DCkFpLLPhjN1XWe7YOkZ9pKLYJUvHg3ufyhgpENvlmd6IQ=
```
Comment (Avantol13) [inline .travis.yml]: what is this exactly? is this safe to be on a public repo?

File: .travis.yml [hunk 4]
```diff
@@ -9,16 +11,37 @@ matrix:
...
+
+deploy:
+  provider: pypi
+  user: uc-ctds
+  skip_existing: true
+  skip_cleanup: true
+  on:
+    tags: true
+  password:
+    secure: OmFLXX6b+AiqPH195KUyndz8HPxQ2rridbVH3y8cDPSaceZWKkR7H6filyuG+CczUQyqXJLxLyfIbI0LDwhxYm6SpSAA7z6mg7thuv1QD+r2dS5yC9/gJAsxqyyJNwdIvviLc8cSAXcTSlL6L1zHoZ3bx7qv8WJ2tMJsSsBLxEg/XFi4nxJ1PAHdrhusPAhx9SAGoocV+ooEQqYVRG+xKfsI5le+sEC9r48eob3oq1SLZ7rJMGOjyNmLpLZiY6Mk6tOQZgDGBSyGhJnER9OzbM1RnJaUuHL23prfePZrqdaoGoorxUJX8gaewkp79c9Mj+utQXfu0+NrJFhsh5y1sXGQAiT2ypJkz3rS+wLVIIuCuIC40WcBxoMcgYpNduTOi10Bnc1tpIC+bQDGt1jEgofb2QxdqJRBNreyYTC+TJFSvWjLZSoETTdPtSJakEspoKTDsKeZvcm/9DXOTMGNphjbYbQm9d0ZZ7eg3EtMtCLtTWwSeUM3hYeKVLrMg2uBp6K54HDJgPH8qGJXDiQHdiP49BiAYkGOVlX3vhqkhXz4xeRk6a6iAzdqy5vT0QrtVaWki2XTcj/upCkEY0uyXTDV9y4OC//mZ8GyeBIRS3blC66E6WF8Y3wtPphkfbwzANELVCl4YPGFGraZiJBpC6zzsfY+gNBthcLDjsAvR6o=
```
Comment (Avantol13) [inline .travis.yml]: same question, this isn't secure credentials that shouldn't be shared, right?

File: release_notes.sh (hunk 2 above)
Comment (fantix) [reply release_notes.sh]: Oh this is set in `.travis.yml`.

File: .travis.yml (hunk 3 above)
Comment (fantix) [reply .travis.yml]: It's the above `GH_TOKEN`. Yes, it is supposed to be encrypted salted with repo slug, timestamp, and host (travis.org or travis.com).

File: .travis.yml (hunk 4 above)
Comment (fantix) [reply .travis.yml]: This is safe I believe: https://docs.travis-ci.com/user/deployment/pypi/

File: README.md (hunk 1 above)
Comment (fantix) [reply README.md]: Oh, I removed this because I didn't find usage in this package.

**Pull request merged**